src/mono/mono/metadata/gc.c

```diff
@@ -702,7 +702,9 @@ mono_gc_finalize_notify (void)
 	if (mono_gc_is_null ())
 		return;
 
-#ifdef HOST_WASM
+#if defined(HOST_WASI)
+	// TODO: Schedule the background job on WASI. Threads aren't yet supported in this build.
```

Comment:
Is this the right approach? Will the objects leak forever because the finalizer hasn't run and then cause the application to fail? I know there are many objects that have finalizers and this seems like it risks breaking code that uses them, unless we're sure everything will behave correctly here as long as (for example) you always Dispose them before they get GC'd

Reply:
It's definitely not the right approach for a complete implementation, but allows us to move forwards with prototyping. It's not expected that all existing code will all work with this PR alone, but it's a step towards that.

Reply:
Would it be valid to just run them synchronously at this point? Then the code generally won't break, I think.

Reply:
I did try exactly that originally and basic scenarios ended up in some kind of deadlock which was hard to diagnose. To be honest I think there are other areas that would be even more critical for making nontrivial code work properly (e.g., async support, and other common platform calls) so didn't think it wise to hold back prototyping on this aspect.